src/enumo/workload.rs

```diff
@@ -120,7 +120,13 @@ impl Workload {
     }
 
     pub fn plug(self, name: impl Into<String>, workload: &Workload) -> Self {
-        Workload::Plug(Box::new(self), name.into(), Box::new(workload.clone()))
+        match workload {
+            // Empty plug is the same as filter excludes
+            Workload::Set(xs) if xs.is_empty() => {
+                self.filter(Filter::Excludes(name.into().parse().unwrap()))
```

Comment:
not a comment on code but we should update paper with this optimization I think. Very cool!